### ISSUE TITLE
Consider unremovable LOCK objects

### ISSUE THREAD

**Issue by alexvanin** [nspcc-dev/neofs-s3-gw#381]
LOCK objects are not stable in NeoFS yet. As for neofs-node v0.28.0-rc.1 they can be removed. However this might be invalid behavior and LOCK objects with expiration header will not be removable by `object.Delete` request. We need to adopt that as soon as behavior will be changed in NeoFS.

**Comment by alexvanin** [comment]
LOCK issues in general are blocked by https://github.com/nspcc-dev/neofs-node/issues/1226 and https://github.com/nspcc-dev/neofs-node/issues/1227

**Comment by alexvanin** [comment]
It seems like we handle unremovable LOCK just fine, according to https://github.com/nspcc-dev/neofs-node/pull/1461#issuecomment-1144890461, so we don't have to do any additional work.

To clarify:
- retention compliance should work fine with unremovable NeoFS LOCKs
- retention governance won't allow to remove lock, because NeoFS LOCK objects are unremovable
- legal hold will be possible after https://github.com/nspcc-dev/neofs-contract/issues/247

**Comment by KirillovDenis** [comment]
Now we can put one lock in `compliance` mode and several locks in `governance` mode (they don't affect each other). Before retention period is end we cannot delete neither lock or object version. After retention period is end object can be successfully deleted. 
Everything seems to be ok when [this branch](https://github.com/nspcc-dev/neofs-node/tree/fix/correct-lock-objects) to be merged into master.
